### PR TITLE
Add keyword `check_occu: bool = True` to `CifParser.get_structures()`

### PR DESCRIPTION
**Pull request by jonathanjdenney**
If `check_occu=False`, site occupancy will not be checked, allowing unphysical occupancy != 1. Useful for experimental results in which occupancy was allowed to refine to unphysical values. Warning: unphysical site occupancies are incompatible with many pymatgen features.

### CODE REVIEW

**Comment by janosh** [comment]
Could you move the test into the existing [`pymatgen/io/tests/test_cif.py`](https://github.com/materialsproject/pymatgen/blob/master/pymatgen/io/tests/test_cif.py)? And add the new kwarg to the doc string? Would be great to have a sentence in there explaining when this might be useful.

**Comment by jonathanjdenney** [comment]
Hello @janosh, My most resent pull request has been sitting for a while and not running any tests. It says it needs maintainer approval. Regarding your doc string comment. There is an explanation in the get_structures doc string. There isn't one in the _get_structure doc string as none of its other arguments were explained there either.

File: pymatgen/io/cif.py
Comment (janosh) [inline pymatgen/io/cif.py:1005]: It looks like there's a lot of code duplication between the `if` and `else` cases here. Without having looked into the details, this seems avoidable here. We aim for DRY code as much as possible.

File: pymatgen/io/tests/test_cif.py
Comment (janosh) [inline pymatgen/io/tests/test_cif.py:1183]: Please use pascel case for clases: `TestSkipChecks`

File: pymatgen/io/tests/test_cif.py
Comment (janosh) [inline pymatgen/io/tests/test_cif.py:1194]: Not needed.

File: pymatgen/io/tests/test_cif.py
Comment (janosh) [reply pymatgen/io/tests/test_cif.py:1183]: This should probably not be its own test class but part of the `CifParserTest`.

**Review by janosh — CHANGES_REQUESTED** [review]

**Comment by jonathanjdenney** [comment]
@janosh, I've addressed the changes you wanted. It says it needs approval before running the tests again. Do you know why sometimes it runs the tests automatically and sometimes it doesn't?

**Comment by janosh** [comment]
> @janosh, I've addressed the changes you wanted.

Thanks!

> Do you know why sometimes it runs the tests automatically and sometimes it doesn't?

GitHub doesn't auto trigger workflows on PRs from 1st time contributors as a security measure.

**Comment by jonathanjdenney** [comment]
@janosh I'm not sure what happened with my most recent commit. It failed in the linting step and is giving over 450 errors, mostly in files I've not worked with at all. Do you have any suggestions on the best way to proceed with this?

**Comment by janosh** [comment]
If you merge the `master` branch into your PR, the errors should disappear.

**Comment by jonathanjdenney** [comment]
@janosh This is waiting for approval again. I think the issues have been resolved.

**Comment by jonathanjdenney** [comment]
@janosh waiting for approval to run the tests.

**Comment by shyuep** [comment]
It fails because you did not commit your `Skip_checks_test.cif` file to the repo. Naturally, that file does not existing on Github.....

File: pymatgen/io/cif.py
Comment (janosh) [inline pymatgen/io/cif.py:1129]: white space

File: pymatgen/io/cif.py
Comment (janosh) [inline pymatgen/io/cif.py:1111]: I think a more descriptive name would be `skip_occu_checks`.

File: test_files/Skip_checks_test.cif
Comment (janosh) [inline test_files/Skip_checks_test.cif:1]: I don't think we actually need a new test file. We can also test this loading an existing CIF test file using `open('some_file.cif')` and then modify the occupancies in that string before passing it to `CifParser.from_string`.

**Review by janosh — CHANGES_REQUESTED** [review]

File: pymatgen/io/cif.py
Comment (jonathanjdenney) [reply pymatgen/io/cif.py]: immediately following that line is a for loop that sets occupancies to 1 if they have a greater than 1 total occupancy and are within the set tolerance. For the skip checks to work I need the unedited version of allspecies.

File: test_files/Skip_checks_test.cif
Comment (jonathanjdenney) [reply test_files/Skip_checks_test.cif:1]: I hope it’s okay to do this, but I’m going to address all the remaining requested changes in this reply before I push the commit. 

1. I’ve added hints to the _get_structures function. Did you want me to add hints to the get_structures function as well or anywhere else for that matter?
2. I renamed skip_checks to skip_occu_checks everywhere I use it and it is defaulted to False. 
3. I am not sure what you mean about your white space comment. Can you clarify?
4. Skip_checks_test.cif has been removed. It now tests site_type_symbol_test.cif and changes the Te occupancy to 1.5000.

Is there anything else I am missing here before I hit go on my next commit?

**Comment by jonathanjdenney** [comment]
@janosh looks like it is waiting for approval again.

File: pymatgen/io/cif.py
Comment (janosh) [inline pymatgen/io/cif.py:953]: Can you add a code comment explaining what's going on here?

**Review by janosh — CHANGES_REQUESTED** [review]
@jonathanjdenney You're close! I fixed your test, improved the doc string and variable names a bit. Now just a few `mypy` errors left for you to fix.

**Comment by jonathanjdenney** [comment]
@janosh I believe everything is working as it should now. I added the description you wanted as well.

File: pymatgen/io/cif.py
Comment (janosh) [inline pymatgen/io/cif.py:1113]: This looks needlessly complicated. What's the reason for creating `struct_2`?

File: pymatgen/io/cif.py
Comment (janosh) [inline pymatgen/io/cif.py:1136]: Same here. Can this be simpler?

**Review by janosh — CHANGES_REQUESTED** [review]

File: pymatgen/io/cif.py
Comment (jonathanjdenney) [reply pymatgen/io/cif.py]: I think the reason struct_2 was there is no longer in cif.py. I've updated the file so that it doesn't make a new object.

**Review by janosh — APPROVED** [review]
Thanks for persisting on this PR @jonathanjdenney! 👍

**Pull request merged**
